lib/certgen/certgen.go

```diff
@@ -260,13 +263,47 @@ func genSANExtension(userName string, kerberosRealm *string) (*pkix.Extension, e
 	return &sanExtension, nil
 }
 
+// stringToIntSlice converts a string with period-separated integers into a
+// slice of integers.
+func stringToIntSlice(input string) ([]int, error) {
+	splitInput := strings.Split(input, ".")
+	if len(splitInput) < 1 {
+		return nil, errors.New("no period separators")
+	}
+	output := make([]int, 0, len(splitInput))
+	for _, strValue := range splitInput {
+		if value, err := strconv.Atoi(strValue); err != nil {
+			return nil, err
+		} else {
+			output = append(output, value)
+		}
+	}
+	return output, nil
+}
+
+func getGroupListExtension(groups []string) (*pkix.Extension, error) {
+	encodedValue, err := asn1.Marshal(groups)
+	if err != nil {
+		return nil, err
+	}
+	oid, err := stringToIntSlice(constants.GroupListOID)
+	if err != nil {
+		return nil, err
+	}
+	groupListExtension := pkix.Extension{
+		Id:    oid,
```

Comment:
I dont like the import of all dominator constants here. Just add a local constant of the oid string text. You remove 1 dependency and about 20 lines of code.

Reply:
Grumble. It's like writing the same value into different databases.
Anyway, dependency removed.